src/main/java/com/pinterest/secor/uploader/S3UploadManager.java

```diff
@@ -41,10 +44,15 @@
  * default credential provider chain (supports environment variables,
  * system properties, credientials file, and IAM credentials).
  *
+ * If set, it will
```

Comment:
I'll have to open another PR I'm afraid. I had to make my fork private to integrate it with our build system and only realised I couldn't continue with this PR afterwards. I've made a new fork but there appears to be no way to link these up. Will close this now.